bundles/org.openhab.binding.mqtt.homie/src/main/java/org/openhab/binding/mqtt/homie/internal/handler/HomieThingHandler.java

```diff
@@ -104,6 +105,14 @@ public void initialize() {
         super.initialize();
     }
 
+    @Override
+    public void handleRemoval() {
+        this.stop();
+        if(config.removetopics)
+            this.removeRetainedTopics();
+        updateStatus(ThingStatus.REMOVED);
```

Comment:
I think you can call super.handleRemoval() instead?

Reply:
Ummmm... "super.handleRemoval()" instead of "updateStatus(ThingStatus.REMOVED)", right? Yeah, that seems fine!